subprojects/core/src/main/java/org/gradle/internal/classpath/intercept/CallInterceptor.java

```diff
@@ -63,20 +42,9 @@ protected CallInterceptor(InterceptScope... interceptScopes) {
      * @return the value to return to the caller
      * @throws Throwable if necessary to propagate it to the caller
      */
-    public abstract Object doIntercept(Invocation invocation, String consumer) throws Throwable;
-
-    MethodHandle decorateMethodHandle(MethodHandle original, MethodHandles.Lookup caller, int flags) {
-        MethodHandle spreader = original.asSpreader(Object[].class, original.type().parameterCount());
-        MethodHandle decorated = MethodHandles.insertArguments(INTERCEPTOR, 0, this, spreader, flags, caller.lookupClass().getName());
-        return decorated.asCollector(Object[].class, original.type().parameterCount()).asType(original.type());
-    }
+    Object doIntercept(Invocation invocation, String consumer) throws Throwable;
```

Comment:
❓ `doIntercept()` is a pattern that suggests a higher-level `intercept()` method. Can we call this `intercept()` instead?

Reply:
I think we can rename it

Reply:
Fixed